apisix/plugins/datadog.lua

```diff
@@ -144,6 +149,12 @@ function _M.log(conf, ctx)
     entry.balancer_ip = ctx.balancer_ip or ""
     entry.route_name = ctx.route_name or ""
     entry.scheme = ctx.upstream_scheme or ""
+    entry.prefer_name = conf.prefer_name
+    -- capture service name at request response cycle if prefer_name is set.
+    if entry.prefer_name and entry.service_id and entry.service_id ~= "" then
+       local svc = service_fetch(entry.service_id)
+       entry.service_name = svc and svc.value.name or ""
```

Comment:
```suggestion
       entry.service_id = svc and svc.value.name or ""
```
So there is no need to check the service_name in the report process.

And we can do the same thing with the route, so there is no need to pass the prefer_name configuration.

Reply:
Ack. Done. Thank you.